allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/behavior_ecephys_project_cloud_api.py

```diff
@@ -0,0 +1,336 @@
+            (default: None)
+        """
+        if manifest_name is None:
+            self.cache.load_last_manifest()
+        else:
+            self.cache.load_manifest(manifest_name)
+
+        expected_metadata = set(["behavior_sessions",
+                                 "sessions",
+                                 "probes",
+                                 "units",
+                                 "channels"])
+
+        if self.cache._manifest.metadata_file_names is None:
+            raise RuntimeError("S3CloudCache object has no metadata "
+                               "file names. BehaviorProjectCloudApi "
+                               "expects a S3CloudCache passed which "
+                               "has already run load_manifest()")
+        cache_metadata = set(self.cache._manifest.metadata_file_names)
+
+        if cache_metadata != expected_metadata:
+            raise RuntimeError("expected S3CloudCache object to have "
+                               f"metadata file names: {expected_metadata} "
+                               f"but it has {cache_metadata}")
+
+        if not self.skip_version_check:
+            data_sdk_version = [i for i in self.cache._manifest._data_pipeline
+                                if i['name'] == "AllenSDK"][0]["version"]
+            version_check(self.cache._manifest.version, data_sdk_version)
+
+        self.logger = logging.getLogger("BehaviorEcephysProjectCloudApi")
+        self._get_ecephys_session_table()
+        self._get_behavior_session_table()
+        self._get_unit_table()
+        self._get_probe_table()
+        self._get_channel_table()
+
+    @classmethod
+    def from_s3_cache(cls, cache_dir: Union[str, Path],
+                      bucket_name: str,
+                      project_name: str,
+                      ui_class_name: str) -> "BehaviorEcephysProjectCloudApi":
+        """instantiates this object with a connection to an s3 bucket and/or
+        a local cache related to that bucket.
+
+        Parameters
+        ----------
+        cache_dir: str or pathlib.Path
+            Path to the directory where data will be stored on the local system
+
+        bucket_name: str
+            for example, if bucket URI is 's3://mybucket' this value should be
+            'mybucket'
+
+        project_name: str
+            the name of the project this cache is supposed to access. This
+            project name is the first part of the prefix of the release data
+            objects. I.e. s3://<bucket_name>/<project_name>/<object tree>
+
+        ui_class_name: str
+            Name of user interface class (used to populate error messages)
+
+        Returns
+        -------
+        BehaviorProjectCloudApi instance
+
+        """
+        cache = S3CloudCache(cache_dir,
+                             bucket_name,
+                             project_name,
+                             ui_class_name=ui_class_name)
+        return cls(cache)
+
+    @staticmethod
```

Comment:
If you make this (and the equivalent method in `BehaviorProjectCloudApi` a `classmethod`, you can do the same trick you did in `from_s3_cache` to and just have this method instantiate `cls`, rather than having it explicitly instantiate `BehaviorEcephysProjectCloudApi`.

Reply:
done